### PR TITLE
implement hid_get_input_report for Linux hidraw backend

### PR DESCRIPTION
**Pull request by Be-ing**
Fixes https://github.com/libusb/hidapi/issues/259

### CODE REVIEW

**Comment by Be-ing** [comment]
I don't know if this actually works with Linux < 5.11, but at least it should build.

**Comment by Be-ing** [comment]
Looking closer at the [commit that implemented this in Linux](https://github.com/torvalds/linux/commit/f43d3870cafa2a0f3854c1819c8385733db8f9ae#), simply defining the macro won't do anything in Linux < 5.11 without the other changes in that commit. So I changed this to keep the old dummy code that just returns an error for old versions of Linux.

**Comment by JoergAtGithub** [comment]
The ifdef checks the Kernel at compile time and returns the error if it was compiled on a too old Kernel.
But what happens if the executable is compiled on a new kernel, but executed on an old one?

**Comment by Be-ing** [comment]
I have not tested that.

**Comment by Be-ing** [comment]
I suppose that could actually happen with Flatpak... I'll test what happens with a VM running an old kernel.

**Comment by Youw** [comment]
Lets have the functionality compile regardless of the kernel version (e.g. if not defined HIDIOCGINPUT - define it);
As fo running on the older kernel - I'm pretty sure the call will simply fail, since the correspondind ioctl isn't supported.

**Comment by Be-ing** [comment]
Okay I switched it back to the original implementation copying the definition from hidraw.h.

File: linux/hid.c [hunk 1]
```diff
@@ -1065,13 +1065,21 @@ int HID_API_EXPORT hid_get_feature_report(hid_device *dev, unsigned char *data,
 	return res;
 }
 
-// Not supported by Linux HidRaw yet
+// HIDIOCGINPUT is not defined in Linux kernel headers < 5.11.
+// This definition is from hidraw.h in Linux >= 5.11.
+// https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=f43d3870cafa2a0f3854c1819c8385733db8f9ae
+#ifndef HIDIOCGINPUT
+#define HIDIOCGINPUT(len)    _IOC(_IOC_WRITE|_IOC_READ, 'H', 0x0A, len)
+#endif
```
Comment (Youw) [inline linux/hid.c:1073]: move it to line 64 of this file

**Review by Youw — APPROVED** [review]
This looks good.
I'd like to check it locally with my device(s) befor having it merged.

**Comment by Be-ing** [comment]
Have you had a chance to test this yet?

**Review by Youw — APPROVED** [review]
I finally had.
Everything looks perfect. Thanks!

**Comment by Be-ing** [comment]
Thanks. Do you have a plan when you will make the next release?

**Comment by Youw** [comment]
The major (minor actually) release is not going to happen soon - nothing really done for that.
But I'll try to make a patch release soon.

**Comment by Be-ing** [comment]
A patch release soon with this fix would be good.

**Comment by Be-ing** [comment]
Any further updates about a patch release?

**Comment by Be-ing** [comment]
Thanks! I made a [PR to update vcpkg](https://github.com/microsoft/vcpkg/pull/22304).

**Comment by Be-ing** [comment]
I also made a [PR to update the Rust bindings](https://github.com/ruabmbua/hidapi-rs/pull/75).

**Pull request merged**